src/core/tools/common/asio_net_provider.cpp

```diff
@@ -132,27 +134,26 @@ namespace dsn {
         void asio_udp_provider::send_message(message_ex* request)
         {
             // prepare parser as there will be concurrent send-message-s
-            auto pr = get_message_parser_info();
-            auto parser_place = alloca(pr.second);
-            auto parser = pr.first(parser_place, _message_buffer_block_size, true);
-
-            auto count = parser->get_send_buffers_count(request);
-            std::unique_ptr<message_parser::send_buf[]> bufs(new message_parser::send_buf[count]);
-            parser->prepare_buffers_on_send(request, 0, bufs.get());
-            // end using parser
-            parser->~message_parser();
+            auto parser = new_message_parser(request->hdr_format);
```

Comment:
we intentionally use on-stack memory to save cost. Any particular reasons for the change?